plugin/hover.py

```diff
@@ -82,18 +82,25 @@ def handle_response(self, response: 'Optional[Any]', point) -> None:
             all_content += self.diagnostics_content(point_diagnostics)
 
         all_content += self.hover_content(point, response)
-        all_content += self.symbol_actions_content()
+        all_content += self.symbol_actions_content(point)
 
         _test_contents.clear()
         _test_contents.append(all_content)  # for testing only
         self.show_hover(point, all_content)
 
-    def symbol_actions_content(self):
+    def symbol_actions_content(self, point):
         actions = []
-        # TODO: filter by client capabilities
-        actions.append("<a href='{}'>{}</a>".format('definition', 'Definition'))
-        actions.append("<a href='{}'>{}</a>".format('references', 'References'))
-        actions.append("<a href='{}'>{}</a>".format('rename', 'Rename'))
+        (x, y) = self.view.rowcol(point)
+        event = {"x": x, "y": y}
+        if self.has_client_with_capability('definitionProvider') and is_at_word(self.view, event):
```

Comment:
I would suggest not including `is_at_word` in every statement but just doing it once. 

But even more so, this check is already done before the hover request is made (see `is_likely_at_symbol`) so perhaps we can avoid doing the same check a different way again?

Reply:
@tomv564  so you are saying `is_at_word` check can be ignore and the client capability check is enough ?

Reply:
Changes are good to the first part of my comment!

But did you see that we already check if the cursor is at a symbol? Is there a need to check it again here?

Reply:
@tomv564 you are right, no need to check